kite-data/kite-data-core/src/main/java/org/kitesdk/data/Datasets.java

```diff
@@ -303,13 +510,19 @@
   }
 
   /**
-   * Delete a {@link Dataset} identified by the given dataset URI.
+   * Delete a {@link Dataset} identified by the given dataset URI. Kite takes care
+   * of any housekeeping, including deleting data and any metadata stored separately.
```

Comment:
When called with a dataset URI, both data and metadata will be deleted. After this method is called, the dataset will no longer exist, unless an exception is thrown.

When called with a view URI, data in that matches the view will be deleted. The dataset's metadata is not changed. This may throw `UnsupportedOperationException` if the delete requires additional work. See the [`View#deleteAll docs`](http://kitesdk.org/docs/0.17.1/apidocs/org/kitesdk/data/View.html#deleteAll()) for more information that should be covered here.